### PR TITLE
Various fixes and changes to build utilities

### PR DESCRIPTION
**Pull request by mcaskill**
### Utilities

Moved exports to end of file to improve readability.

Separated generic helpers from build helpers:

1. Moved `build/utils/*.js` to `build/helpers/*.js`.
2. From `build/helpers/config.js`:
   - Moved functions `merge` and `isObjectLike` to `build/utils/index.js`.
3. From `build/helpers/template.js`:
   - Moved functions `flatten` and `escapeRegExp` to `build/utils/index.js`.
4. From `build/tasks/styles.js`:
   - Moved and renamed function `createPostCSSProcessor` to `build/helpers/postcss.js` as `createProcessor`.
5. Replaced occurrences of `Object.assign` with `merge` for task options parsing in all tasks.

### Tasks

- Changed behaviour of `includes` to accept a string that will be cast to an array; in `concats.js`, `scripts.js`, and `svgs.js`.

### Glob

- Added constant `supportsGlob` to provide a boolean to check if a glob function is available.
- Fixed inconsistent API support between `tiny-glob`, `globby`, and `glob`, to match `fast-glob`.
- Fixed broken support for `fileOnly` preset options for `tiny-glob`.

### PostCSS

- Added constant `supportsPostCSS` to provide a boolean to check if PostCSS is available.

### General

- Added and improved block comments and type definitions across tasks and utilities.

### CODE REVIEW

File: build/helpers/config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    merge(loconfig, usrconfig);
+} catch (err) {
+    // do nothing
+}
+
+export default loconfig;
+
+export {
+    loconfig,
```
Comment (devenini) [inline build/helpers/config.js:22]: `loconfig` is already exported. And merge export is missing.
```suggestion
    merge,
```

File: build/helpers/config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+try {
+    const usrconfig = await import('../../loconfig.local.json', {
+        assert: { type: "json" },
+    });
+    usrconfig = usrconfig.default;
+
+    merge(loconfig, usrconfig);
+} catch (err) {
+    // do nothing
+}
```
Comment (devenini) [inline build/helpers/config.js:17]: For some reason, the `usrconfig` constant object is not being merged properly. I suggest changing the `usrconfig` constant object back into a `let` statement to make it mutable and able to be merged with the other object.

```suggestion
let usrconfig;

try {
    usrconfig = await import('../../loconfig.local.json', {
        assert: { type: 'json' }
    });
    usrconfig = usrconfig.default;

    merge(loconfig, usrconfig);
} catch (err) {
    // do nothing
}
```

**Review by devenini — DISMISSED** [review]
I had some issues running the `start` task, but with the following suggestions I manage to make it work properly 👍

File: build/helpers/config.js (hunk 1 above)
Comment (mcaskill) [reply build/helpers/config.js:24]: I've resolved this by having `watch.js` import `merge` from `utils`.

File: build/helpers/config.js (hunk 2 above)
Comment (mcaskill) [reply build/helpers/config.js]: I've reverted this section to use `let` instead of `const`.

**Review by devenini — APPROVED** [review]

**Pull request merged**